modules/rhpam-apb/roles/generate-secret/tasks/main.yml

```diff
@@ -32,9 +32,66 @@
               service: '{{ service_name }}'
           data: "{{ lookup('template', 'secret.yml.j2') | from_yaml }}"
   when:
+    - not amq_secret
     - secret_cmd.rc != 0
     - state == 'present'
 
+- name: "[AMQ] Generate secret if it does not exist"
```

Comment:
In my opinion, you are creating a certificate almost in the same way as for the general case among other things. So that part should be reusable. Let me know if I understood correctly...
1. Generate a key pair for the broker into broker keystore
1. Generate a key pair for the client into client keystore
1. Export the broker public key to import it to the client keystore
1. Export the client public key to import it to the broker keystore
1. Create the secret with both keystores
1. Use the keystore and truststore in the same DC

If this is correct, the AMQ pods will serve the broker cert and trust the client cert and vice-versa. Why not just generate one cert and use it as both keystore and truststore? If the user provides a different truststore and keystore then use it but for self-generated certs that should be enough. WDYT?

Reply:
The problem about the keystore generated for AMQ is that, user will need to have the the client.ts and client.ks that is also generated. To make possible to user to re-use the generated files we'll have to add to the generated secret the values for the client.ks and ts plus the password used (as annotation).